src/main/java/com/keniu/services/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.keniu.services;
+
+import com.keniu.dto.UserLoginDto;
+import com.keniu.dto.UserLoginRequestDto;
+import com.keniu.security.JwtUtil;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private final JwtUtil jwtUtil;
+    private final AuthenticationManager authenticationManager;
+
+    @Override
+    public UserLoginDto authenticate(UserLoginRequestDto userLoginRequestDto) {
+        authenticationManager.authenticate(
+            new UsernamePasswordAuthenticationToken(
+                userLoginRequestDto.getEmail(),
+                userLoginRequestDto.getPassword()
+            )
+        );
+        return new UserLoginDto(
+            jwtUtil.generateToken(userLoginRequestDto.getEmail()));
```

Comment:
Let's make it a little easier to read. This could also be useful if we need to log a generated token or something like that
```suggestion
String token = jwtUtil.generateToken(userLoginRequestDto.getEmail());
        return new UserLoginDto(token);
```

Reply:
Added